lib/storage-handler.js

```diff
@@ -29,14 +31,14 @@ class StorageHandler {
 
       this.storage.setItem('LAST_REPO_UPDATE', lastCheck);
 
-      let existingRepositories = this.storage.getItem('REPOSITORIES') || [];
-      let allRepositories = existingRepositories.concat(newRepositories);
-      debug('sorting repositories according to date..');
-      allRepositories.sort((a, b) => {
-        return new Date(b.created_at) - new Date(a.created_at);
-      });
+      let allRepositories = newRepositories;
+      if(allRepositories.length < MAX_RETURN_ROWS_NEW) {
+        debug('adding existing repositories back');
+        let existingRepositories = this.storage.getItem('REPOSITORIES') || [];
+        allRepositories = allRepositories.concat(existingRepositories);
```

Comment:
I think it would make sense to add the slice here, as it's covering the same use case, what do you think?

Reply:
The slice has to apply to both cases, where we need the old repos and where we don't need them. It doesn't help to do the slice any earlier than before storing. We still have to fetch all the new repos and possibly fetch all the existing repos.

Reply:
true!